SCons/Platform/__init__.py

```diff
@@ -256,7 +259,8 @@ def __call__(self, target, source, env, for_signature):
         else:
             tempfile_dir = None
 
-        fd, tmp = tempfile.mkstemp(suffix, dir=tempfile_dir, text=True)
+        # default is binary - encode the tempfile contents later
```

Comment:
Seems wrong to open the file as text when we're explicitly going to write bytes. Risks unwanted conversion?